File: lib/queue.js
var job = require('./db.js')
var encode = require('./encode.js');
/**
 * Start Processing Encoding Job Queue
 **/

exports.process = function(callback) {
        job.find('queued', function(jobs) {
            jobs.forEach(function(job) {
                console.log("processing " + job.id + job.original_file_path);
                switch (job.request_formats) {
                    case 'mp4':
                        console.log('video #' + job.id)
                        encode.video(job);
                        break
                    case 'mp3':
                        encode.audio(job);
                        break
                    case 'thumbnails':
                        encode.thumbnails(job);
                        break
                    default:
                        console.log("request_formats is not set for Job #" + job.id)
                callback(job)
                }
            })
        })
    } //end of processQueue

File: app/queue.py
from .models import Job
from .encoder import Encoder
from pprint import pprint
from app import queue_db


class Queue():

    def process(self, status="queued"):
        jobs = queue_db.session.query(Job).filter_by(status=status)
        for job in jobs:
            pprint(vars(job))
            e = Encoder(job.file_path, job.thumb_count)
            if job.request_format == 'thumbnails':
                e.thumbs()
            elif job.request_format == 'mp3':
                e.audio()
            else:
                e.video()
        return "jobs processed successfully"
